src/middlewared/middlewared/plugins/account_/privilege.py

```diff
@@ -0,0 +1,260 @@
+import enum
+import errno
+
+from middlewared.schema import accepts, Bool, Dict, Int, List, Ref, Str, Patch
+from middlewared.service import CallError, CRUDService, private, ValidationErrors
+from middlewared.service_exception import MatchNotFound
+import middlewared.sqlalchemy as sa
+
+
+class BuiltinPrivileges(enum.Enum):
+    LOCAL_ADMINISTRATOR = "LOCAL_ADMINISTRATOR"
+
+
+class PrivilegeModel(sa.Model):
+    __tablename__ = "account_privilege"
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+    builtin_name = sa.Column(sa.String(200), nullable=True)
+    name = sa.Column(sa.String(200))
+    local_groups = sa.Column(sa.JSON(type=list))
+    ds_groups = sa.Column(sa.JSON(type=list))
+    allowlist = sa.Column(sa.JSON(type=list))
+    web_shell = sa.Column(sa.Boolean())
+
+
+class PrivilegeService(CRUDService):
+
+    keys = {}
+
+    class Config:
+        namespace = "privilege"
+        datastore = "account.privilege"
+        datastore_extend = "privilege.item_extend"
+        datastore_extend_context = "privilege.item_extend_context"
+        cli_namespace = "auth.privilege"
+
+    ENTRY = Dict(
+        "privilege_entry",
+        Int("id"),
+        Str("builtin_name", null=True),
+        Str("name", required=True, empty=False),
+        List("local_groups", items=[Int("local_group")]),
+        List("ds_groups", items=[Int("ds_group")]),
+        List("allowlist", items=[Ref("allowlist_item")]),
+        Bool("web_shell", required=True),
+    )
+
+    @private
+    async def item_extend_context(self, rows, extra):
+        return {
+            "groups": await self._groups(),
+        }
+
+    @private
+    async def item_extend(self, item, context):
+        item["local_groups"] = self._local_groups(context["groups"], item["local_groups"])
+        item["ds_groups"] = await self._ds_groups(context["groups"], item["ds_groups"])
+        return item
+
+    @accepts(Patch(
+        "privilege_entry",
+        "privilege_create",
+        ("rm", {"name": "builtin_name"}),
+    ))
+    async def do_create(self, data):
+        """
+        Creates a privilege.
+
+        `name` is a name for privilege.
+
+        `local_groups` is a list of local user account groups that gain this privilege.
```

Comment:
Should probably specify that these are GIDS (so that people don't think it's the DB IDs.